packages/request-logic/src/request-logic.ts

```diff
@@ -59,6 +60,21 @@ export default class RequestLogic implements RequestLogicTypes.IRequestLogic {
       hashedTopics,
     );
 
+    // Create nft for erc20 nft payment network
+    if (this.transactionManager.tokenizeRequest && requestParameters.extensionsData) {
+      const paymentData = requestParameters.extensionsData[0];
+      if (paymentData.id === ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_NFT_CONTRACT) {
+        if (requestParameters.payee) {
+          const recipient = requestParameters.payee.value; // payee is the owner of nft
+          const assetToken = requestParameters.currency.value;
+          const reqIdObj = MultiFormat.deserialize(requestId);
+          const tokenId = reqIdObj.value; // tokenId is requestId.value
+          const metadata = Buffer.from(reqIdObj.type).toString('base64'); // metadata is requestId.type
+          await this.transactionManager.tokenizeRequest(recipient, assetToken, tokenId, metadata);
```

Comment:
Please don't make `request-logic` initiate the NFT minting

Instead, the app layer should create the Request (call request-client) and mint the NFT (call payment-processor) in two separate calls.

Reply:
Will do